src/edu/csus/ecs/pc2/ui/ShadowControlPane.java

```diff
@@ -1042,10 +1089,27 @@ public void updateShadowNumberofRecords(int nRec)
             lastRecordTextfield.setText(String.valueOf(numRecord));
         }
     }
-    
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void updateShadowNumberofTossedRecords(int nRec)
+    {
+        // if the number of tossed records is different from what we last display and it's valid
+        // update the instrumentation.
+        if(nRec != numTossedRecord && nRec >= 0) {
+            // TODO: Do we want to "InvokeLater" this (simple) update to a text field?
+            numTossedRecord = nRec;
+            // Save to file? Send to server contestinfo?
+            lastTossedRecordTextfield.setText(String.valueOf(numTossedRecord));
```

Comment:
if this is not already on the Swing thread, then yes it should be moved to the Swing Thread.

Reply:
I was concerned since this (and the one above it, **updateShadowNumberofRecords()**) changes frequently and very fast, would there be an EXTRA penalty if we do an InvokeLater? Or, would the numbers not update in a timely fashion? Delayed?  Since it's just a 'setText()' is this problem?

Reply:
a bit of a delay is better than a deadlock.

Reply:
Put the setText()'s in invokelater blocks.  Tested by running NAC2024 Systest1 shadowing my test server.  No noticeable delays in updating the GUI text fields.  I did confirm that the setText()'s were NOT on the AWT thread, so it was wrong as it was (it was that way for quite some time - I do recall some "freezes" of the GUI and some exceptions over the past couple of years - but it was during debugging, so it could have been related to this).  Anyway, as a CI I fixed the 3 sections of code that called setText() directly.  Pushed in latest commit.